controllers/managedocs_controller.go

```diff
@@ -690,86 +690,27 @@ func (r *ManagedOCSReconciler) reconcileAlertmanagerConfigSecret() error {
 			return fmt.Errorf("DeadMan's Snitch secret does not contain a SNITCH_URL entry")
 		}
 
-		alertmanagerConfig := r.generateAlertmanagerConfig(pagerdutyServiceKey, dmsURL)
-		config, err := yaml.Marshal(alertmanagerConfig)
-		if err != nil {
-			return fmt.Errorf("Unable to encode alertmanager conifg: %v", err)
-		}
-		r.alertmanagerConfigSecret.Data = map[string][]byte{
-			"alertmanager.yaml": config,
+		desired := templates.AlertmanagerConfigTemplate.DeepCopy()
+		for i := range desired.Spec.Receivers {
+			receiver := &desired.Spec.Receivers[i]
+			switch receiver.Name {
+			case "pagerduty":
+				receiver.PagerDutyConfigs[0].ServiceKey.Key = "PAGERDUTY_KEY"
+				receiver.PagerDutyConfigs[0].ServiceKey.LocalObjectReference.Name = r.PagerdutySecretName
+				receiver.PagerDutyConfigs[0].Details[0].Key = "SOP"
+				receiver.PagerDutyConfigs[0].Details[0].Value = r.SOPEndpoint
+			case "DeadMansSnitch":
+				receiver.WebhookConfigs[0].URL = &dmsURL
+			}
 		}
+		r.alertmanagerConfig.Spec = desired.Spec
+		r.alertmanagerConfig.ObjectMeta.Labels = map[string]string{"dhruv": "test"}
```

Comment:
@bindrad  is that here by mistake? if so what should be the actual matcher? 
Also, if it is static it should go into the template

Reply:
It was a mistake, I and Pulkit were trying to contribute to each other's PR.